moveit_ros/visualization/src/moveit_ros_visualization/moveitjoy_module.py

```diff
@@ -523,15 +523,13 @@ def parseSRDF(self):
         planning_groups = {}
         for g in ri.get_group_names():
             self.planning_groups_tips[g] = ri.get_group_joint_tips(g)
-            planning_groups[g] = [
-                "/rviz/moveit/move_marker/goal_" + l
-                for l in self.planning_groups_tips[g]
-            ]
+            if len(self.planning_groups_tips[g]) > 0:
+                planning_groups[g] = [
+                    "/rviz/moveit/move_marker/goal_" + l
+                    for l in self.planning_groups_tips[g]
+                ]
         for name in planning_groups.keys():
-            if len(planning_groups[name]) == 0:
-                del planning_groups[name]
-            else:
-                print(name, planning_groups[name])
+            print(name, planning_groups[name])
```

Comment:
Nit: I bet that fixes the error nicely but "while there", why not write it like this:
```
for group_name, planning_groups in planning_groups:
    print(group_name, planning_groups)
```
Either way, looks good to me.